aws/rust-runtime/aws-runtime/src/auth/sigv4.rs

```diff
@@ -141,28 +143,23 @@ impl SigV4Signer {
             }
         }
     }
-}
 
-impl Sign for SigV4Signer {
-    fn sign_http_request(
+    /// Signs the given `request`.
+    ///
+    /// This is a helper used by [`Sign::sign_http_request`] and will be useful if calling code
+    /// needs to pass a configured `settings`.
+    pub fn sign_http_request(
```

Comment:
We should think about this function signature some more. A lot of customers want to do manual signing for use cases such as API Gateway invocations, and having a function that is easy to sign with would help a lot for that. If this requires `RuntimeComponents` or `ConfigBag`, it becomes more difficult to use directly.

Reply:
Added `TODO` based on our offline discussion (573ef36). Will circle back!